src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -828,6 +828,13 @@ impl NamespacedOrchestrator for NamespacedKubernetesOrchestrator {
                     ..Default::default()
                 }],
                 volumes,
+                security_context: Some(PodSecurityContext {
+                    // clusterd and environmentd both always run as gid 999, as defined
+                    // in their `Dockerfile`'s. This change is only relevant when there is
+                    // a volume mount, but we apply is unconditionally.
+                    fs_group: Some(999),
+                    ..Default::default()
+                }),
```

Comment:
This violates the `KubernetesOrchestrator` abstraction boundary. The `KubernetesOrchestrator` is not meant to have any knowledge of `environmentd` or `clusterd`, and it shouldn't be hardcoding group IDs that come from a far away Dockerfile!

I think the best thing to do is to add an optional `--orchestrator-kubernetes-service-fs-group` CLI argument to environmentd that gets plumbed into here and is set appropriately by the cloud infrastructure. This has the nice benefit that it won't impact the cloudtest tests, which apparently don't need this set.

Reply:
this is a good point, i can pipe that through!